src/webapp/pages/form-page/disease-outbreak-event/useDiseaseOutbreakEventForm.ts

```diff
@@ -0,0 +1,704 @@
+import { useCallback, useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+
+import { Maybe } from "../../../../utils/ts-utils";
+import i18n from "../../../../utils/i18n";
+import { useAppContext } from "../../../contexts/app-context";
+import { Id } from "../../../../domain/entities/Ref";
+import { FormState } from "../../../components/form/FormState";
+import { FormFieldState } from "../../../components/form/FormFieldState";
+import { FormType } from "../FormPage";
+import { DiseaseOutbreakEvent } from "../../../../domain/entities/DiseaseOutbreakEvent";
+
+export type GlobalMessage = {
+    text: string;
+    type: "warning" | "success" | "error";
+};
+
+export type DiseaseOutbreakEventFormStateLoaded = {
+    kind: "loaded";
+    data: FormState;
+};
+
+export type DiseaseOutbreakEventFormStateLoading = {
+    kind: "loading";
+};
+
+export type DiseaseOutbreakEventFormStateError = {
+    kind: "error";
+    message: string;
+};
+
+export type DiseaseOutbreakEventFormState =
+    | DiseaseOutbreakEventFormStateLoaded
+    | DiseaseOutbreakEventFormStateLoading
+    | DiseaseOutbreakEventFormStateError;
+
+export function useDiseaseOutbreakEventForm(
+    formType: FormType,
+    diseaseOutbreakEventId?: Id
+): {
+    globalMessage: Maybe<GlobalMessage>;
+    formState: DiseaseOutbreakEventFormState;
+    handleFormChange: (newFormState: FormState, updatedField: FormFieldState) => void;
+    onSaveForm: () => void;
+    onCancelForm: () => void;
+} {
+    const { compositionRoot } = useAppContext();
+    const history = useHistory();
+
+    const [globalMessage, setGlobalMessage] = useState<Maybe<GlobalMessage>>();
+    const [formState, setFormState] = useState<DiseaseOutbreakEventFormState>({ kind: "loading" });
+    const [diseaseOutbreakEvent, setDiseaseOutbreakEvent] = useState<DiseaseOutbreakEvent>();
+
+    useEffect(() => {
+        if (diseaseOutbreakEventId) {
```

Comment:
Yes, I agree that when there is an id then we get populated form and if not thenn just the empty form. I will change this in the next PR joining presentation with domain because this hook was just an aproximation but it is going to change 100% with the union of the other layers.

Reply:
makes sense, thanks Ana.